src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitedArray = date.split(fromFormat[fromFormat.length - 1]);
+  let year = '';
+  const separator = toFormat[toFormat.length - 1];
+
+  toFormat.pop();
+
+  for (const i of toFormat) {
+    if (i === 'DD') {
+      toFormat[toFormat.indexOf(i)] = splitedArray[fromFormat.indexOf(i)];
+    }
+
+    if (i === 'MM') {
+      toFormat[toFormat.indexOf(i)] = splitedArray[fromFormat.indexOf(i)];
+    }
+
+    if ((i === 'YYYY' || i === 'YY') && fromFormat.includes(i)) {
+      year = splitedArray[fromFormat.indexOf(i)];
+      toFormat[toFormat.indexOf(i)] = year;
+    } else if (i === 'YY') {
+      year = splitedArray[fromFormat.indexOf('YYYY')].slice(2);
+      toFormat[toFormat.indexOf(i)] = year;
+    } else if (i === 'YYYY') {
+      if (Number(splitedArray[fromFormat.indexOf('YY')]) < 30) {
+        year = '20' + splitedArray[fromFormat.indexOf('YY')];
+      } else {
+        year = '19' + splitedArray[fromFormat.indexOf('YY')];
+      }
+      toFormat[toFormat.indexOf(i)] = year;
+    }
+  }
```

Comment:
Repeatedly calling `indexOf` inside the loop is inefficient because it traverses the array multiple times. It would be more efficient to calculate the indexes once before the loop and then use them inside the loop.